src/DemoData.hs

```diff
@@ -120,58 +124,62 @@ data CommentInContext = CommentInContext
     , _cicComment :: Comment
     }
 
-genComment :: [Idea] -> [User] -> forall m . PersistM m => Gen (m CommentInContext)
+genComment :: [Idea] -> [User] -> forall m . ActionM m => Gen (m CommentInContext)
 genComment ideas students = do
     (idea, student) <- ideaStudentPair ideas students
-    fmap (CommentInContext idea Nothing) .
-        addCommentToIdea (idea ^. _Id) . (,) student <$> arbDocument
+    let event = AddCommentToIdea (idea ^. _Id)
+        getResult = fmap (CommentInContext idea Nothing)
+    getResult . addWithUser event student <$> arbDocument
 
-genReply :: [CommentInContext] -> [User] -> forall m . PersistM m => Gen (m CommentInContext)
+genReply :: [CommentInContext] -> [User] -> forall m . ActionM m => Gen (m CommentInContext)
 genReply comments_in_context students = do
     CommentInContext idea Nothing comment <- elements comments_in_context
     (_, student) <- ideaStudentPair [idea] students
-    fmap (CommentInContext idea (Just comment)) .
-        addReplyToIdeaComment (idea ^. _Id) (comment ^. _Id) . (,) student <$> arbDocument
+    let event = AddReplyToIdeaComment (idea ^. _Id) (comment ^. _Id)
+        getResult = fmap (CommentInContext idea (Just comment))
+    getResult . addWithUser event student <$> arbDocument
 
-genCommentVote :: [CommentInContext] -> [User] -> forall m . PersistM m => Gen (m CommentVote)
+genCommentVote :: [CommentInContext] -> [User] -> forall m . ActionM m => Gen (m CommentVote)
 genCommentVote comments_in_context students = do
     CommentInContext idea mparent comment <- elements comments_in_context
     (_, student) <- ideaStudentPair [idea] students
-    case mparent of
-        Nothing ->
-            addCommentVoteToIdeaComment (idea ^. _Id) (comment ^. _Id) . (,) student <$> arb
-        Just parent ->
-            addCommentVoteToIdeaCommentReply (idea ^. _Id) (parent ^. _Id) (comment ^. _Id) . (,) student <$> arb
+    let action = case mparent of
+            Nothing ->
+                addWithUser $ AddCommentVoteToIdeaComment
+                    (idea ^. _Id) (comment ^. _Id)
+            Just parent ->
+                addWithUser $ AddCommentVoteToIdeaCommentReply
+                    (idea ^. _Id) (parent ^. _Id) (comment ^. _Id)
+    action student <$> arb
 
-updateAvatar :: User -> URL -> forall m . PersistM m => m ()
-updateAvatar user url = modifyUser (user ^. _Id) (userAvatar ?~ url)
+updateAvatar :: User -> URL -> forall m . ActionM m => m ()
+updateAvatar user url = aupdate $ SetUserAvatar (user ^. _Id) url
 
 
 -- * Universe
 
-mkUniverse :: forall m . PersistM m => IO (m ())
-mkUniverse = do
-    r <- newQCGen
-    return (universe r)
+mkUniverse :: forall m . ActionM m => IO (m ())
+mkUniverse = universe <$> newQCGen
 
-universe :: QCGen -> forall m . PersistM m => m ()
-universe rnd = void $ do
+universe :: QCGen -> forall m . ActionM m => m ()
```

Comment:
This type change will generate a lot of transactions. This should be run in only one. That's why the demo data generation is takes too much time.